### PR TITLE
use main field over module for runtime

### LINKED ISSUE #3434: vuejs: render ssr
### What version of Bun is running?

0.6.11

### What platform is your computer?

Darwin 22.5.0 x86_64 i386

### What steps can reproduce the bug?

> Official docs: https://vuejs.org/guide/scaling-up/ssr.html#rendering-an-app

```ts
// this runs in Node.js on the server.
import { createSSRApp } from "vue";
// Vue's server-rendering API is exposed under `vue/server-renderer`.
import { renderToString } from "vue/server-renderer";

const app = createSSRApp({
    data: () => ({ count: 1 }),
    template: `<button @click="count++">{{ count }}</button>`,
});

renderToString(app).then((html) => {
    console.log(html);
});

```

### What is the expected behavior?

```html
<button>1</button>
```

### What do you see instead?

```html
<!---->
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by dylan-conway**
fixes #3434

### CODE REVIEW

File: src/resolver/resolver.zig [hunk 1]
```diff
@@ -3408,7 +3408,10 @@ pub const Resolver = struct {
                             // with this same path. The goal of this code is to avoid having
                             // both the "module" file and the "main" file in the bundle at the
                             // same time.
-                            if (kind != ast.ImportKind.require) {
+                            //
+                            // Additionally, if this is for the runtime, use the "main" field.
+                            // If it doesn't exist, the "module" field will be used.
+                            if (!r.opts.runtime and kind != ast.ImportKind.require) {
```
Comment (Jarred-Sumner) [inline src/resolver/resolver.zig]: Since this is only used by the Resolver, maybe we could make this a flag inside Resolver like `enable_module_field`?

Comment (Jarred-Sumner) [reply src/resolver/resolver.zig]: and then default it to true but set it to false in VirtualMachine.init

Comment (dylan-conway) [reply src/resolver/resolver.zig]: good idea, updated

File: src/bun.js/javascript.zig [hunk 2]
```diff
@@ -898,6 +898,7 @@ pub const VirtualMachine = struct {
 
         vm.bundler.macro_context = null;
         vm.bundler.resolver.store_fd = store_fd;
```
Comment (Jarred-Sumner) [inline src/bun.js/javascript.zig:901]: What about the initStandaloneModuleGraph function?

**Pull request merged**